File: test/VbNet.LanguageServer.Tests/Services/SemanticTokensServiceTests.cs
using Microsoft.Extensions.Logging.Abstractions;
using VbNet.LanguageServer.Protocol;
using VbNet.LanguageServer.Services;
using VbNet.LanguageServer.Workspace;
using Xunit;

namespace VbNet.LanguageServer.Tests.Services;

/// <summary>
/// Unit tests for SemanticTokensService.
/// </summary>
public class SemanticTokensServiceTests
{
    private readonly WorkspaceManager _workspaceManager;
    private readonly DocumentManager _documentManager;
    private readonly SemanticTokensService _semanticTokensService;

    public SemanticTokensServiceTests()
    {
        _workspaceManager = new WorkspaceManager(NullLogger<WorkspaceManager>.Instance);
        _documentManager = new DocumentManager(_workspaceManager, NullLogger<DocumentManager>.Instance);
        _semanticTokensService = new SemanticTokensService(
            _workspaceManager,
            _documentManager,
            NullLogger<SemanticTokensService>.Instance);

        _workspaceManager.Initialize();
    }

    [Fact]
    public async Task GetSemanticTokensAsync_NoDocument_ReturnsEmpty()
    {
        var result = await _semanticTokensService.GetSemanticTokensAsync(new SemanticTokensParams
        {
            TextDocument = new TextDocumentIdentifier { Uri = "file:///nonexistent.vb" }
        }, CancellationToken.None);

        Assert.NotNull(result);
        Assert.Empty(result.Data);
    }

    [Fact]
    public async Task GetSemanticTokensAsync_NullParams_ReturnsEmpty()
    {
        var result = await _semanticTokensService.GetSemanticTokensAsync(null!, CancellationToken.None);

        Assert.NotNull(result);
        Assert.Empty(result.Data);
    }
}

File: test/VbNet.LanguageServer.Tests.Vb/Services/SemanticTokensServiceTests.vb
Imports System.Threading
Imports System.Threading.Tasks
Imports Microsoft.Extensions.Logging.Abstractions
Imports VbNet.LanguageServer.Protocol
Imports VbNet.LanguageServer.Services
Imports VbNet.LanguageServer.Workspace
Imports Xunit

Namespace VbNet.LanguageServer.Tests.Services

    ''' <summary>
    ''' Unit tests for SemanticTokensService.
    ''' </summary>
    Public Class SemanticTokensServiceTests
        Private ReadOnly _workspaceManager As WorkspaceManager
        Private ReadOnly _documentManager As DocumentManager
        Private ReadOnly _semanticTokensService As SemanticTokensService

        Public Sub New()
            _workspaceManager = New WorkspaceManager(NullLogger(Of WorkspaceManager).Instance)
            _documentManager = New DocumentManager(_workspaceManager, NullLogger(Of DocumentManager).Instance)
            _semanticTokensService = New SemanticTokensService(
                _workspaceManager,
                _documentManager,
                NullLogger(Of SemanticTokensService).Instance)

            _workspaceManager.Initialize()
        End Sub

        <Fact>
        Public Async Function GetSemanticTokensAsync_NoDocument_ReturnsEmpty() As Task
            Dim result = Await _semanticTokensService.GetSemanticTokensAsync(New SemanticTokensParams With {
                .TextDocument = New TextDocumentIdentifier With {.Uri = "file:///nonexistent.vb"}
            }, CancellationToken.None)

            Assert.NotNull(result)
            Assert.Empty(result.Data)
        End Function

        <Fact>
        Public Async Function GetSemanticTokensAsync_NullParams_ReturnsEmpty() As Task
            Dim result = Await _semanticTokensService.GetSemanticTokensAsync(Nothing, CancellationToken.None)

            Assert.NotNull(result)
            Assert.Empty(result.Data)
        End Function
    End Class

End Namespace
